### DISCUSSION TITLE
Displaying heading tags and pasting code in codeblock tag

### DISCUSSION THREAD

**Discussion by jhefreyzz** [outline/rich-markdown-editor#237] [General] [answered]
Hi,

I play around the outline web app since it was on the github trending repositories. I'm planning to use the markdown editor for my hobby project, if possible can I ask the following questions.

1. How does the editor displays the header tag after editing
  ` h1 | This is heading tag.`, 
   I looked at the source code to know how it was implemented but I wasn't able to found it but I learned that the text was injected through ::before pseudo
  `.gxkOKc h1:not(.placeholder)::before { content: "h1"}`

2. How do I paste a piece of code in a codeblock tag?
  For example I have an existing codeblock tag with code in it, then I paste another code inside it. I expect that it will merge the code but what happens is that the newly pasted code creates a new codeblock.

If the questions I asked are not valid, you can close this discussion.
Thank you and more power.

**Comment by tommoor** [comment, marked as answer]
1. It's not ideal, but that's part of the display css for read only here: 
https://github.com/outline/rich-markdown-editor/blob/develop/src/index.tsx#L719-L740

2. How do I paste a piece of code in a codeblock tag?
It's quite difficult to do, I think this could be counted as a bug.
